lib/github_changelog_generator/parser.rb

```diff
@@ -224,51 +214,80 @@ def self.setup_parser(options)
         end
       end
     end
+  end
 
-    # @return [Options] Default options
-    def self.default_options
-      Options.new(
-        date_format: "%Y-%m-%d",
-        output: "CHANGELOG.md",
-        base: "HISTORY.md",
-        issues: true,
-        add_issues_wo_labels: true,
-        add_pr_wo_labels: true,
-        pulls: true,
-        filter_issues_by_milestone: true,
-        issues_of_open_milestones: true,
-        author: true,
-        unreleased: true,
-        unreleased_label: "Unreleased",
-        compare_link: true,
-        exclude_labels: ["duplicate", "question", "invalid", "wontfix", "Duplicate", "Question", "Invalid", "Wontfix", "Meta: Exclude From Changelog"],
-        summary_labels: ["Release summary", "release-summary", "Summary", "summary"],
-        breaking_labels: ["backwards-incompatible", "Backwards incompatible", "breaking"],
-        enhancement_labels: ["enhancement", "Enhancement", "Type: Enhancement"],
-        bug_labels: ["bug", "Bug", "Type: Bug"],
-        deprecated_labels: ["deprecated", "Deprecated", "Type: Deprecated"],
-        removed_labels: ["removed", "Removed", "Type: Removed"],
-        security_labels: ["security", "Security", "Type: Security"],
-        configure_sections: {},
-        add_sections: {},
-        issue_line_labels: [],
-        max_issues: nil,
-        simple_list: false,
-        ssl_ca_file: nil,
-        verbose: true,
-        header: "# Changelog",
-        merge_prefix: "**Merged pull requests:**",
-        issue_prefix: "**Closed issues:**",
-        summary_prefix: "",
-        breaking_prefix: "**Breaking changes:**",
-        enhancement_prefix: "**Implemented enhancements:**",
-        bug_prefix: "**Fixed bugs:**",
-        deprecated_prefix: "**Deprecated:**",
-        removed_prefix: "**Removed:**",
-        security_prefix: "**Security fixes:**",
-        http_cache: true,
-        require: []
-      )
+  class Parser
+    class << self
+      def parse_options(argv = ARGV)
+        options = default_options
+
+        argv_parser = ArgvParser.new(options)
+
+        argv_parser.parse!(argv) # parse first to get config_file
+        FileParserChooser.new(options, options[:config_file]).parse!
+        argv_parser.parse!(argv) # parse again to override whatever was read from the config file
```

Comment:
Do you mean to make the parameters even more generic to the parsers like: `FileParserChooser.new(options).parse!`

Then the parser can choose what/if it will do and the caller don't really need to care about special parameters.

Reply:
Took some inspiration from this comments and changed the parsing totally from the first proposed version.